include/jemalloc/internal/edata.h

```diff
@@ -289,7 +289,47 @@ edata_szind_get(const edata_t *edata) {
 
 static inline size_t
 edata_usize_get(const edata_t *edata) {
-	return sz_index2size(edata_szind_get(edata));
+	assert(edata != NULL);
+	/*
+	 * When sz_limit_usize_gap_enabled() is true, two cases:
+	 * 1. if usize_from_ind is not smaller than SC_LARGE_MINCLASS,
+	 * usize_from_size is accurate;
+	 * 2. otherwise, usize_from_ind is accurate.
+	 *
+	 * When sz_limit_usize_gap_enabled() is not true, the two should be the
+	 * same when usize_from_ind is not smaller than SC_LARGE_MINCLASS.
+	 */
+	szind_t szind = edata_szind_get_maybe_invalid(edata);
```

Comment:
We can also call `edata_szind_get` like we did before, `double_free` will be getting an assertion when `config_debug` is enabled, and the same output if debug is disabled. I changed to calling `maybe_invalid` because it will not give the assertion but instead lead us to a failure in the `safety_check` because `usize` is an invalid value 0. 

I understand that this is risky because we are not failing early/ASAP while the benefit is to get a more easy-to-understand output for invalid inputs. I am a bit hesitant to do so. WDYT? @interwq